dbt-snowflake/src/dbt/adapters/snowflake/connections.py

```diff
@@ -114,13 +114,22 @@ class SnowflakeCredentials(Credentials):
     # this needs to default to `None` so that we can tell if the user set it; see `__post_init__()`
     reuse_connections: Optional[bool] = None
     s3_stage_vpce_dns_name: Optional[str] = None
+    oauth_authorization_url: Optional[str] = None
+    oauth_token_request_url: Optional[str] = None
+    oauth_redirect_uri: Optional[str] = None
+    oauth_scope: Optional[str] = None
+    oauth_disable_pkce: Optional[bool] = None
+    oauth_enable_refresh_token: Optional[bool] = None
+    oauth_enable_single_use_refresh_tokens: Optional[bool] = None
 
     def __post_init__(self):
-        if self.authenticator != "oauth" and (self.oauth_client_secret or self.oauth_client_id):
+        if self.authenticator not in ("oauth", "OAUTH_AUTHORIZATION_CODE") and (
```

Comment:
we'll probably want this to be lower snake case 🙏 

Again @amychen1776 do we have this named anywhere internally?